tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -76,6 +78,89 @@ int main(int argc, char **argv)
 
         EXPECT_EQUAL(count, S2N_CIPHER_SUITE_COUNT);
 
+        {
+            /* Test that the client allows the server to select ciphers that were offered in ClientHello */
+            {
+                conn->client_protocol_version = S2N_TLS13;
+                conn->actual_protocol_version = S2N_TLS13;
+                conn->server_protocol_version = S2N_TLS13;
+
+                /* The client will offer the default tls13 ciphersuites */
+                EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "default_tls13"));
+
+                /* The server will send a TLS13 cipher over the wire */
+                uint8_t valid_wire_ciphers[] = {
+                    TLS_AES_128_GCM_SHA256
+                };
+
+                /* We expect to succeed because the cipher was offered by the client */
+                EXPECT_SUCCESS(s2n_set_cipher_as_client(conn, valid_wire_ciphers));
+
+                EXPECT_SUCCESS(s2n_connection_wipe(conn));
+            }
+
+            /* Test that the client rejects a cipher that was not originally offered in ClientHello */
+            {
+                conn->client_protocol_version = S2N_TLS13;
+                conn->actual_protocol_version = S2N_TLS13;
+                conn->server_protocol_version = S2N_TLS13;
+
+                /* The client will offer the default tls13 ciphersuites */
+                EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "test_all_tls13"));
+
+                /* The server will send a TLS12 cipher over the wire */
+                uint8_t invalid_wire_ciphers[] = {
+                    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                };
+
+                /* We expect to fail because the cipher was not offered by the client */
+                EXPECT_FAILURE_WITH_ERRNO(s2n_set_cipher_as_client(conn, invalid_wire_ciphers), S2N_ERR_CIPHER_NOT_SUPPORTED);
+
+                EXPECT_SUCCESS(s2n_connection_wipe(conn));
+            }
+        }
+
+        {
```

Comment:
What are these {} for? How are these tests related, and why do they need to be separated further?